src/voice/udp/backends/tokio.rs

```diff
@@ -0,0 +1,37 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::net::SocketAddr;
+
+use crate::errors::VoiceUdpError;
+
+#[derive(Debug, Clone)]
+pub struct TokioBackend;
+
+pub type TokioSocket = tokio::net::UdpSocket;
+
+impl TokioBackend {
+    pub async fn connect(url: SocketAddr) -> Result<TokioSocket, VoiceUdpError> {
+        // Bind with a port number of 0, so the os assigns this listener a port
```

Comment:
> So the os assigns this listener a port

Is this true only for Linux, or does this work on macOS and Windows as well?

Reply:
It should be true regardless of OS, according to the [tokio docs](https://docs.rs/tokio/latest/tokio/net/struct.UdpSocket.html#method.bind)